2386/2/ParticipantService.rb

```diff
@@ -0,0 +1,45 @@
+require_relative 'Counters'
+require_relative 'Participant'
+
+# :reek:UtilityFunction
+# Participant Service
+class ParticipantService
+  PATH_FOLDER = 'Rapbattle'.freeze
+  def add_participants(participant)
+    participants_titles = find_participants_titles(participant)
+    Participant.new(hash(participant, participants_titles))
+  end
+
+  def hash(participant, participants_titles)
+    {
+      name: participant,
+      battles:  participants_titles.size,
+      bad_words: bad_words(participant),
+      bad_in_round: bad_in_round(participant, participants_titles),
+      words_in_round: words_in_round(participant)
+    }
+  end
+
+  def bad_words(participant)
+    Counters.count_bad(find_participants_titles(participant))
+  end
+
+  def bad_in_round(participant, participants_titles)
+    (bad_words(participant).to_f / participants_titles.size).round(2)
```

Comment:
it's better to perform rounding and another formatting when u draw the table. Always separate data processing from presentation.